contracts/compounder/rewardHandlers/BalancerSwapsHandler.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.11;
+
+import { SafeERC20 } from "@openzeppelin/contracts-0.8/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts-0.8/token/ERC20/IERC20.sol";
+import { IBalancerVault, IAsset } from "../../interfaces/balancer/IBalancerCore.sol";
+import { IRewardHandler } from "../../interfaces/balancer/IRewardHandler.sol";
+import { HandlerBase } from "./HandlerBase.sol";
+
+contract BalancerSwapsHandler is HandlerBase {
+    using SafeERC20 for IERC20;
+
+    // ----------------------------------------------------------------
+    // Types
+    // ----------------------------------------------------------------
+
+    struct SwapPath {
+        bytes32[] poolIds;
+        address[] assetsIn;
+    }
+    address public immutable AURA_TOKEN;
+    address public immutable AURABAL_TOKEN;
+    SwapPath internal swapPath;
+
+    constructor(
+        address _token,
+        address _strategy,
+        address _balancerVault,
+        address _weth,
+        address _aura,
+        address _auraBal,
+        SwapPath memory _swapPath
+    ) HandlerBase(_token, _strategy, _balancerVault, _weth) {
+        AURA_TOKEN = _aura;
+        AURABAL_TOKEN = _auraBal;
+        _validateSwapPath(_swapPath, _token, _aura, _auraBal);
+        swapPath = _swapPath;
+    }
+
+    function _validateSwapPath(
+        SwapPath memory _swapPath,
+        address _token,
+        address _aura,
+        address _auraBal
+    ) internal view returns (bool) {
+        address[] memory assetsIn = _swapPath.assetsIn;
+        uint256 assetsInLength = assetsIn.length;
+
+        require(assetsInLength > 0, "!poolIds");
+        require(_swapPath.poolIds.length == assetsIn.length, "parity");
+        require(_token != AURA_TOKEN, "token=AURA");
+        require(_token == assetsIn[0], "!swap path");
+        require(_token != AURABAL_TOKEN, "token=AURABAL");
+
+        for (uint256 i = 0; i < assetsInLength; i++) {
+            require(assetsIn[i] != AURABAL_TOKEN, "token=AURABAL");
+        }
+    }
+
+    function getSwapPath() external view returns (SwapPath memory) {
```

Comment:
Is it possible to just make `swapPath` public?

Reply:
No, I get this error on hardhat 

TypeError: Internal or recursive type is not allowed for public state variables.
  --> contracts/compounder/rewardHandlers/BalancerSwapsHandler.sol:24:5:
   |
24 |     SwapPath public swapPath;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^